src/main/InstanceService.ts

```diff
@@ -258,23 +259,53 @@ class InstanceService {
     const instance = this.getInstance(instanceId);
 
     if (instance.account.service) {
-      return instance.client.client.api.getClients();
+      return instance.account.service.client.getClients();
     } else {
       throw new Error(`Account service for instance ${instanceId} not set.`);
     }
   }
 
-  async removeAllOtherClients(instanceId: string, password: string): Promise<void> {
-    const instance = this.getInstance(instanceId);
+  async removeAllClients(backend: string, email: string, password: string): Promise<void> {
+    const backendType = backend === 'staging' ? APIClient.BACKEND.STAGING : APIClient.BACKEND.PRODUCTION;
+    const engine = new MemoryEngine();
+    await engine.init('temporary');
+    const apiClient = new APIClient({store: engine, urls: backendType});
+    const account = new Account(apiClient);
 
-    if (instance.account.service) {
-      const clients = await instance.client.client.api.getClients();
-      for (const client of clients) {
-        await instance.client.client.api.deleteClient(client.id, password);
+    const ClientInfo: ClientInfo = {
+      classification: ClientClassification.DESKTOP,
+      cookieLabel: 'default',
+      model: `E2E Test Server v${version}`,
+    };
+
+    const loginData = {
+      clientType: ClientType.PERMANENT,
+      email,
+      password,
+    };
+
+    try {
+      await account.login(loginData, true, ClientInfo);
+    } catch (error) {
```

Comment:
The function only deletes clients when it catches the login error. otherwise it just logs in and out again.